### PR TITLE
[Merged by Bors] - fix(Order/Zorn): update usage example to Lean 4 syntax

### PR DESCRIPTION
**Pull request by grunweg**
Double-checking the syntax is welcome.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Order/Zorn.lean [hunk 1]
```diff
@@ -32,27 +32,25 @@ This file comes across as confusing to those who haven't yet used it, so here is
...
-  let s : Set α := {x | whatever x},
-  suffices : ∃ x ∈ s, ∀ y ∈ s, y ⊆ x → y = x, -- or with another operator
-  { exact proof_post_zorn },
-  apply zorn_subset, -- or another variant
-  rintro c hcs hc,
-  obtain rfl | hcnemp := c.eq_empty_or_nonempty, -- you might need to disjunct on c empty or not
-  { exact ⟨edge_case_construction,
+lemma zorny_lemma : zorny_statement := by
+  let s : Set α := {x | whatever x}
+  suffices : ∃ x ∈ s, ∀ y ∈ s, y ⊆ x → y = x from -- or with another operator xxx
```
Comment (eric-wieser) [inline Mathlib/Order/Zorn.lean:43]:
```suggestion
  suffices ∃ x ∈ s, ∀ y ∈ s, y ⊆ x → y = x by -- or with another operator xxx
```

File: Mathlib/Order/Zorn.lean [hunk 2]
```diff
@@ -32,27 +32,25 @@ This file comes across as confusing to those who haven't yet used it, so here is
 walkthrough:
 1. Know what relation on which type/set you're looking for. See Variants above. You can discharge
   some conditions to Zorn's lemma directly using a `_nonempty` variant.
-2. Write down the definition of your type/set, put a `suffices : ∃ m, ∀ a, m ≺ a → a ≺ m, { ... },`
+2. Write down the definition of your type/set, put a `suffices : ∃ m, ∀ a, m ≺ a → a ≺ m by ...`
```
Comment (eric-wieser) [inline Mathlib/Order/Zorn.lean:35]:
```suggestion
2. Write down the definition of your type/set, put a `suffices ∃ m, ∀ a, m ≺ a → a ≺ m by ...`
```

**Review by eric-wieser — APPROVED** [review]
bors d+

**Comment by grunweg** [comment]
Thank you for double-checking and the fast review.
bors r+

**Pull request closed without merging**